arbnode/transaction_streamer.go

```diff
@@ -1388,6 +1388,42 @@ func (s *TransactionStreamer) getEspressoSubmittedPos() ([]arbutil.MessageIndex,
 	return pos, nil
 }
 
+func (s *TransactionStreamer) getLastPotentialMsg() (*arbostypes.MessageWithMetadata, error) {
+	lastPotentialMsgBytes, err := s.db.Get(lastPotentialMsgInBatch)
+	if err != nil {
+		if dbutil.IsErrNotFound(err) {
+			log.Info("last potential msg is empty, is trying to post a new batch")
+			return nil, nil
+		}
+	}
+
+	var msgWithMetadata arbostypes.MessageWithMetadata
+	err = rlp.DecodeBytes(lastPotentialMsgBytes, &msgWithMetadata)
+
+	if err != nil {
+		return nil, err
+	}
+	return &msgWithMetadata, nil
+}
+
+func (s *TransactionStreamer) getLastPotentialMsgPos() (*arbutil.MessageIndex, error) {
```

Comment:
Is this necessary? We don't have the message pos in the metadata of `MessageWithMetadata` returned from `getLastPotentialMsg`? If so then fine, it's just a little sad

Reply:
Unfortunately we [don't](https://github.com/EspressoSystems/nitro-espresso-integration/blob/17734f4cfdb58c523aa26870ddf9a57c2f328857/arbos/arbostypes/messagewithmeta.go#L17C1-L20C2)